Shift Left: Build and test extension/llm/... buck packages packages whenever possible

mergennachin opened the pull request:
We had a few reverts last week's oncall. 

Having these in place would have prevented them.

Test Plan: 

unittest-buck CI job

kimishpatel reviewed (approved):
Review automatically exported from Phabricator review in Meta.

mergennachin committed:
Shift Left: Build and test extension/llm/... buck packages packages whenever possible (#14542)

Summary:
We had a few reverts last week's oncall.

Having these in place would have prevented them.

Pull Request resolved: https://github.com/pytorch/executorch/pull/14542

Test Plan: unittest-buck CI job

Reviewed By: kimishpatel

Differential Revision: D83148558

Pulled By: mergennachin

mergennachin committed:
Merge branch 'main' into run_buck_build_for_llm
